### PR TITLE
Feature/restoring saved game

### LINKED ISSUE #17: Add unfinished games page
Add a page with information about not finished games with an abilitiy to restore it and continue playing

### PR DESCRIPTION
**Pull request by llmaxxxll**
closed #17

### CODE REVIEW

File: dice-game-ui-angular/src/app/game/game.component.ts [hunk 1]
```diff
@@ -47,7 +47,6 @@ export class GameComponent implements OnInit {
     } catch {
       throw new Error('The game was not started.');
     }
-
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/game/game.component.ts:50]: Why did you remove it? It was nice.

File: dice-game-ui-angular/src/app/nav-menu/nav-menu.component.html [hunk 2]
```diff
@@ -1,6 +1,7 @@
-<mat-toolbar color='primary'>  
-  <mat-toolbar color='primary'>
+<mat-toolbar color="primary">  
+  <mat-toolbar color="primary">
     <span>Dice Game</span>
-    <button mat-flat-button color="primary" [routerLink]="['start']">Start Game</button>
+    <button mat-flat-button color="primary" [routerLink]="'start'">Start New Game</button>
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/nav-menu/nav-menu.component.html:4]: `routerLink` without brackets and then url without quotes.

File: dice-game-ui-angular/src/app/saved-games/saved-games.component.html [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<h1 align="center" class="text-muted">Saved games</h1>
+
+<div *ngIf="games.length>0; else noSavedGames">
+    <mat-card class="m-1" *ngFor="let ge of games">
+        <mat-card-header>
+            <mat-card-title>
+                <button mat-flat-button [routerLink]="['/game', ge.guid]">
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/saved-games/saved-games.component.html:7]: It is better to make a function to get a link from game object.

File: dice-game-ui-angular/src/app/saved-games/saved-games.component.html [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+                    Single game {{ge.gameEngine.fieldSize.width}} x {{ge.gameEngine.fieldSize.height}} from {{ge.creationDate|date:"dd.MM.yyyy"}}
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/saved-games/saved-games.component.html:8]: And it is opposite here - single quotes, because it is TS code, not html.

File: dice-game-ui-angular/src/app/saved-games/saved-games.component.html [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+                </button>
+            </mat-card-title>
+        </mat-card-header>
+        <mat-card-actions align="end">
+            <button mat-mini-fab color="primary" align="start" [routerLink]="['/game', ge.guid]" aria-label="Continue" class="m-1">
+                <mat-icon>login</mat-icon>
+            </button>
+            <button mat-mini-fab color="accent" (click)="removeSavedGame(ge.guid)" aria-label="Remove" class="m-1">
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/saved-games/saved-games.component.html:16]: You'd better pass whole game object to the method.

File: dice-game-ui-angular/src/app/saved-games/saved-games.component.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import { Component, OnInit } from '@angular/core';
+
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/saved-games/saved-games.component.ts:2]: Excessive empty line.

File: dice-game-ui-angular/src/app/saved-games/saved-games.component.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  games: SavedGameEngine[];
+
+  constructor(private gameStorageService: GameStorageService) {}
+
+  ngOnInit(): void {
+    this.games = this.gameStorageService.getAllSavedGames();
+  }
+
+  removeSavedGame(guid: Guid) {
+    alert(guid);
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/saved-games/saved-games.component.ts:22]: Not needed debug code.

File: dice-game-ui-angular/src/app/saved-games/saved-games.component.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    this.gameStorageService.removeGame(guid);
+    this.games = this.gameStorageService.getAllSavedGames();
+    alert(this.games);
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/saved-games/saved-games.component.ts:25]: And this.

File: dice-game-ui-angular/src/app/services/game-storage-service.ts [hunk 9]
```diff
@@ -2,10 +2,24 @@ import { Injectable } from '@angular/core';
 import { Guid } from 'guid-typescript';
 import { GameEngine } from 'engine';
 
+export class SavedGameEngine {
+
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/services/game-storage-service.ts:6]: Excessive empty line.

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts [hunk 10]
```diff
@@ -1,33 +1,70 @@
...
-import { GameEngine, SimpleGameEngine, Size } from 'engine';
-import { GameStorageService } from './game-storage-service';
+import { GameEngine, SimpleGameEngine } from 'engine';
+import { GameStorageService, SavedGameEngine } from './game-storage-service';
 
 @Injectable()
 export class LocalGameStorageService implements GameStorageService {
+  private readonly savedGamesKey = 'saved-games';
+
   createGame(gameEngine: GameEngine): Guid {
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/services/local-game-storage-service.ts:10]: I think you can remove `createGame` method and merge the logic with `saveGame`. It doesn't have much sense to separate these concepts.

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts [hunk 11]
```diff
@@ -1,33 +1,70 @@
...
+      localStorage.setItem(this.savedGamesKey, JSON.stringify(curSavedGames));
+    }
+    return Guid.parse(savedGameEngine.guid);
   }
 
   saveGame(gameEngine: GameEngine, guid: Guid): Guid {
-    if (!localStorage.getItem(guid.toString())) {
-      throw new Error('The saved game was not found in the store');
+    const allSavedGames = this.getAllSavedGames();
+    if (!allSavedGames) {
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/services/local-game-storage-service.ts:22]: This condition is impossible, because the method will always return at least an empty array.

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts [hunk 12]
```diff
@@ -1,33 +1,70 @@
...
+      throw new Error('This game was not created in the store.');
     }
-    localStorage.setItem(guid.toString(), gameEngine.fieldSize.width.toString());
+    const curSavedGame = allSavedGames.filter((savedGame) => savedGame.guid === guid.toString());
+    if (!curSavedGame) {
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/services/local-game-storage-service.ts:26]: `filter` method will always return a value, at least an empty array. You need to use `find` method instead.

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts [hunk 13]
```diff
@@ -1,33 +1,70 @@
...
+    );
+  }
+
+  getAllSavedGames(): SavedGameEngine[] {
+    let curSavedGames: SavedGameEngine[] = [];
+    const savedGames = localStorage.getItem(this.savedGamesKey);
+    if (savedGames) {
+      curSavedGames = (JSON.parse(savedGames) as SavedGameEngine[]) ?? [];
+    }
+    return curSavedGames.sort((game1, game2) => (game1.creationDate < game2.creationDate ? 1 : -1));
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/services/local-game-storage-service.ts:55]: It is not this class duty to sort return games, it is a part of business logic of the application. Thus it should be done by saved games component.

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts [hunk 14]
```diff
@@ -1,33 +1,70 @@
...
+      curSavedGame[0].gameEngine.fieldSize.width,
+      curSavedGame[0].gameEngine.fieldSize.height
+    );
+  }
+
+  getAllSavedGames(): SavedGameEngine[] {
+    let curSavedGames: SavedGameEngine[] = [];
+    const savedGames = localStorage.getItem(this.savedGamesKey);
+    if (savedGames) {
+      curSavedGames = (JSON.parse(savedGames) as SavedGameEngine[]) ?? [];
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/services/local-game-storage-service.ts:53]: Or you can simply return `savedGames ? JSON.parse(savedGames) as SavedGameEngine[] : [];` without `curSavedGames` temporary variable.

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts [hunk 15]
```diff
@@ -1,33 +1,70 @@
...
   }
 
   removeGame(guid: Guid): void {
-    localStorage.removeItem(guid.toString());
+    const curSavedGames = this.getAllSavedGames();
+    const curSavedGame = curSavedGames.filter((savedGame) => savedGame.guid === guid.toString());
+    if (!curSavedGame) {
+      throw new Error(`A game with id: ${guid.toString()} was not found.`);
+    }
+    const index = curSavedGames.indexOf(curSavedGame[0]);
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/services/local-game-storage-service.ts:64]: You can use `findIndex` instead which can find an index by some condition.

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts (hunk 10 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/services/local-game-storage-service.ts:10]: Actually, I did this to evade passing a gameEngine object as a parameter to the function. I decided that passing 'null' is not a good idea while creating a new game. If you don't agree with that, I'll change it.

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts (hunk 11 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/services/local-game-storage-service.ts:22]: Is that not working as a "length === 0" check?

Comment (sponkrashin) [reply dice-game-ui-angular/src/app/services/local-game-storage-service.ts:22]: Definitely not. It checks for null, undefined, 0, empty string and NaN.

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts (hunk 10 above)
Comment (sponkrashin) [reply dice-game-ui-angular/src/app/services/local-game-storage-service.ts:10]: Maybe you misprinted and you meant `Guid` object, because you already passes game engine to both methods.

File: dice-game-ui-angular/src/app/game/game.component.ts (hunk 1 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/game/game.component.ts:50]: What actually did I remove? I couldn't find.

File: dice-game-ui-angular/src/app/saved-games/saved-games.component.html (hunk 3 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/saved-games/saved-games.component.html:7]: I'm not completely understand what you mean. ((

File: dice-game-ui-angular/src/app/nav-menu/nav-menu.component.html (hunk 2 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/nav-menu/nav-menu.component.html:4]: I used that because the docs tells to do that but, no problem, I'll change it.

File: dice-game-ui-angular/src/app/saved-games/saved-games.component.html (hunk 5 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/saved-games/saved-games.component.html:16]: Why? The actual method doesn't need it.

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts (hunk 10 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/services/local-game-storage-service.ts:10]: Yes, of course, I meant guid object.

File: dice-game-ui-angular/src/app/nav-menu/nav-menu.component.html (hunk 2 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/nav-menu/nav-menu.component.html]: Fixed

File: dice-game-ui-angular/src/app/saved-games/saved-games.component.html (hunk 4 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/saved-games/saved-games.component.html]: Got it. Fixed.

File: dice-game-ui-angular/src/app/saved-games/saved-games.component.ts (hunk 6 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/saved-games/saved-games.component.ts]: Fixed

File: dice-game-ui-angular/src/app/saved-games/saved-games.component.ts (hunk 7 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/saved-games/saved-games.component.ts]: My fault, forgot it. Fixed

File: dice-game-ui-angular/src/app/saved-games/saved-games.component.ts (hunk 8 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/saved-games/saved-games.component.ts]: Fixed

File: dice-game-ui-angular/src/app/services/game-storage-service.ts (hunk 9 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/services/game-storage-service.ts]: Removed

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts (hunk 11 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/services/local-game-storage-service.ts]: Gor it. Changed

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts (hunk 12 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/services/local-game-storage-service.ts:26]: ok. Fixed

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts (hunk 13 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/services/local-game-storage-service.ts]: Agree. Moved to saved-game-component - the place where I need it.

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts (hunk 14 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/services/local-game-storage-service.ts]: ok. Changed

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts (hunk 15 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/services/local-game-storage-service.ts]: Got it. Changed the method.

File: dice-game-ui-angular/src/app/game/game.component.ts (hunk 1 above)
Comment (sponkrashin) [reply dice-game-ui-angular/src/app/game/game.component.ts:50]: Empty string.

File: dice-game-ui-angular/src/app/saved-games/saved-games.component.html (hunk 3 above)
Comment (sponkrashin) [reply dice-game-ui-angular/src/app/saved-games/saved-games.component.html:7]: Make a function in a component code that will return a link for the router which will navigate to a game with such guid. Like
```ts
getGameLink(gameEngine: GameEngine) {
  return `/game/${gameEngine.guid}`; // Or return ['/game', gameEngine.guid];
}
```

File: dice-game-ui-angular/src/app/saved-games/saved-games.component.html (hunk 5 above)
Comment (sponkrashin) [reply dice-game-ui-angular/src/app/saved-games/saved-games.component.html:16]: Because method implementation could change and in my case it won't lead to changes in a template, only in component code. But it's up to you.

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts (hunk 10 above)
Comment (sponkrashin) [reply dice-game-ui-angular/src/app/services/local-game-storage-service.ts:10]: You can add a default value to guid param of `saveGame` function which will equal to `null`. And then you will call `saveGame(engine)` from start game component and `saveGame(engine, engine.guid)` in game component.

File: dice-game-ui-angular/src/app/saved-games/saved-games.component.html (hunk 3 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/saved-games/saved-games.component.html:7]: Honestly, It's a really strange and unnatural decision for me. I can stand at least for creating a method which derectly navigate to the page (as a part shifting control to the component) but your decision doesn't look as a good practice for me. Of course, you have more experience here and if you say that would be better, I'll change it.

File: dice-game-ui-angular/src/app/saved-games/saved-games.component.html (hunk 5 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/saved-games/saved-games.component.html:16]: Ok, then let's leave it "as is" because now I don't see any cases when the method removing saved game engine needs the whole object.

File: dice-game-ui-angular/src/app/saved-games/saved-games.component.html (hunk 3 above)
Comment (sponkrashin) [reply dice-game-ui-angular/src/app/saved-games/saved-games.component.html:7]: Common practice is to not have much logic in a template, because it means mixing of presentation and behavior which is a violation of some good programming principles. Your template doesn't need to know how exactly you calculate a game link, it just need to know that it exists.

Comment (sponkrashin) [reply dice-game-ui-angular/src/app/saved-games/saved-games.component.html:7]: For example, in some large projects you can have a single service that manages all navigation links and build needed link with parameters if needed. If you decide to do this in your project, you will just need to change a component method without changing a template.

File: dice-game-ui-angular/src/app/game/game.component.ts (hunk 1 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/game/game.component.ts]: Returned

File: dice-game-ui-angular/src/app/saved-games/saved-games.component.html (hunk 3 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/saved-games/saved-games.component.html]: Ok. I changed.

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts (hunk 10 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/services/local-game-storage-service.ts]: Moved creating to the saveGame

File: dice-game-ui-angular/src/app/services/game-storage-service.ts [hunk 16]
```diff
@@ -2,10 +2,25 @@ import { Injectable } from '@angular/core';
...
 import { GameEngine } from 'engine';
 
+export class SavedGameEngine {
+  constructor(gameEngine: GameEngine) {
+    this.gameEngine = gameEngine;
+    this.guid = Guid.create().toString();
+    this.creationDate = new Date();
+  }
+
+  gameEngine: GameEngine;
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/services/game-storage-service.ts:12]: Fields should be placed before a constructor.

File: dice-game-ui-angular/src/app/services/game-storage-service.ts [hunk 17]
```diff
@@ -2,10 +2,25 @@ import { Injectable } from '@angular/core';
...
 @Injectable()
 export abstract class GameStorageService {
-  abstract createGame(gameEngine: GameEngine): Guid;
-  abstract saveGame(gameEngine: GameEngine, guid: Guid): Guid;
   abstract restoreGame(guid: Guid): GameEngine;
   abstract removeGame(guid: Guid): void;
+  abstract getAllSavedGames(): SavedGameEngine[];
+
+  saveGame(gameEngine: GameEngine, guid: Guid = null): Guid {
+    throw new Error('The method was not implemented');
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/services/game-storage-service.ts:24]: Why haven't you done the same for other abstract methods?

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts [hunk 18]
```diff
@@ -1,33 +1,70 @@
...
-  createGame(gameEngine: GameEngine): Guid {
-    const guid = Guid.create();
-    localStorage.setItem(guid.toString(), gameEngine.fieldSize.width.toString());
-    return guid;
-  }
+  private readonly savedGamesKey = 'saved-games';
+
+  saveGame(gameEngine: GameEngine, guid: Guid = null): Guid {
+    const allSavedGames = this.getAllSavedGames();
+    if (!guid && allSavedGames.length === 0) {
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/services/local-game-storage-service.ts:12]: So, the first game will never be created.

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts [hunk 19]
```diff
@@ -1,33 +1,70 @@
...
+      throw new Error('This game was not created in the store.');
+    }
+
+    let savedGame: SavedGameEngine = allSavedGames.find(
+      (sg) => sg.guid === (guid ? guid.toString() : Guid.EMPTY)
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/services/local-game-storage-service.ts:17]: Use `guid?.toString()` instead, it will be enough I suppose.

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts [hunk 20]
```diff
@@ -1,33 +1,70 @@
...
+    // if saving previously saved game
+    if (guid) {
+      if (!savedGame) {
+        throw new Error('This game was not created in the store.');
+      }
+      savedGame.gameEngine = gameEngine;
+
+      // if saving a new game
+    } else {
+      if (savedGame) {
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/services/local-game-storage-service.ts:28]: Impossible condition. Saved game will never be found if you don't pass a guid to the function.

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts [hunk 21]
```diff
@@ -1,33 +1,70 @@
...
+
+    localStorage.setItem(this.savedGamesKey, JSON.stringify(allSavedGames));
     return guid;
   }
 
   restoreGame(guid: Guid): GameEngine {
-    const fieldSize: number = JSON.parse(localStorage.getItem(guid.toString()));
-    if (!fieldSize) {
+    const allSavedGames = this.getAllSavedGames();
+    if (allSavedGames.length === 0) {
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/services/local-game-storage-service.ts:43]: This all can be done easier.
```ts
const curSavedGame = allSavedGames.find(sg => sg.guid === guid.toString());
return curSavedGame
  ? new SimpleGameEngine(curSavedGame.fieldSize.width, curSavedGame.fieldSize.height)
  : null;
```

File: dice-game-ui-angular/src/app/services/game-storage-service.ts (hunk 17 above)
Comment (sponkrashin) [reply dice-game-ui-angular/src/app/services/game-storage-service.ts:24]: And related question - why have you done this at all?

Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/services/game-storage-service.ts:24]: I guess because abstract methods don't need an implementation ) 
I needed to set a default value for the guid so I had to make this method not abstract and I throw an error as a way to make class's user to implement it

File: dice-game-ui-angular/src/app/services/game-storage-service.ts (hunk 16 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/services/game-storage-service.ts]: Changed

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts (hunk 18 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/services/local-game-storage-service.ts]: Fixed

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts (hunk 19 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/services/local-game-storage-service.ts]: Changed

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts (hunk 20 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/services/local-game-storage-service.ts]: Removed

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts (hunk 21 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/services/local-game-storage-service.ts:41]: Changed

File: dice-game-ui-angular/src/app/services/game-storage-service.ts (hunk 17 above)
Comment (sponkrashin) [reply dice-game-ui-angular/src/app/services/game-storage-service.ts:24]: Oh, I see. It's strange that TS doesn't allow abstract methods to have parameters with default values.

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts [hunk 22]
```diff
@@ -1,33 +1,72 @@
...
-  createGame(gameEngine: GameEngine): Guid {
-    const guid = Guid.create();
-    localStorage.setItem(guid.toString(), gameEngine.fieldSize.width.toString());
-    return guid;
-  }
+  private readonly savedGamesKey = 'saved-games';
+
+  saveGame(gameEngine: GameEngine, guid: Guid = null): Guid {
+    const allSavedGames = this.getAllSavedGames();
+    if (guid && allSavedGames.length === 0) {
```
Comment (sponkrashin) [inline dice-game-ui-angular/src/app/services/local-game-storage-service.ts:12]: This condition is also not needed because you check the same in the line 20. If you passed guid and saved games array is empty then savedGame from line 16 will be undefined.

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts (hunk 21 above)
Comment (sponkrashin) [reply dice-game-ui-angular/src/app/services/local-game-storage-service.ts:41]: Maybe it is better to remove this check and add check that saved game with such guid exists instead of returning null.

File: dice-game-ui-angular/src/app/services/local-game-storage-service.ts (hunk 22 above)
Comment (llmaxxxll) [reply dice-game-ui-angular/src/app/services/local-game-storage-service.ts]: Removed
